shared/src/utils/propositionalLogicMinimize.ts

```diff
@@ -0,0 +1,324 @@
+import {
+  BinaryOperatorType,
+  Literal,
+  NormalForm,
+  Operator,
+  SyntaxTreeNodeType,
+  TruthTable,
+} from "@shared/utils/propositionalLogic.ts"
+
+/**
+ * During the Quine–McCluskey algorithm, the datatype for a truthTable can be a BinaryMix
+ * 0 --> false; 1 --> true; x --> true/false
+ */
+type BinaryMix = 0 | 1 | "x"
+
+/**
+ * This class gets the minimal normal form of a given expression.
+ * - DNF (Disjunctive Normal Form)
+ * - CNF (Conjunctive Normal
+ * - The minimal normal form is the minimal number of literals needed
+ */
+export class GetMinimalNormalForm {
```

Comment:
Just call the class `MinimalNormalForm`. Classes represent a thing, not an action.